src/pages/SingleMatch.tsx

```
@@ -128,6 +139,15 @@ const SingleMatch = () => {
         await refetch();
         setCreateAppointment(false);
     };
+
+    const startAdHocMeeting = async () => {
+        const meetingData = await createAdHocMeeting({ variables: { matchId: matchId } });
+        const appointmentId = meetingData && meetingData.data?.matchCreateAdHocMeeting.id;
+        const appointmentType = meetingData && meetingData.data?.matchCreateAdHocMeeting.appointmentType;
+        await refetch();
+        navigate(`/video-chat/${appointmentId}/${appointmentType}`);
+    };
```

useCallback? And also what if the appointmentType oder Id is undefined?
We should throw an error if so imho. This would then trigger the ErrorBoundary.